src/ocean/math/WideInt.d

```diff
@@ -0,0 +1,474 @@
+/*******************************************************************************
+
+    Implementation of integer-like struct with arbitrary but fixed size, defined
+    via template argument.
+
+    Such struct is a value type with stable binary layout for a given size which
+    is the primary goal of this implementation. Performance may be eventually
+    improved if necessity arises, right now implementation is intentionally
+    simplistic and naive.
+
+    Internally wide integer is represented by a static array of `uint` values
+    such that concatenating their binary representation together results in long
+    binary sequence representing actual stored number.
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.math.WideInt;
+
+import ocean.transition;
+import ocean.core.Test;
+import ocean.core.Enforce;
+
+/// ditto
+public struct WideInt ( size_t N )
+{
+    static assert (
+        N > 2,
+        "For 'N <= 2' using 'uint' or 'ulong' directly is suggested"
+    );
+
+    private uint[N] payload;
```

Comment:
FYI: Some compilers have intrinsics to support these operations. Here are the gcc ones:
https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html
D should really add something like this as well.